src/screens/ImportWalletScreen.tsx

```diff
@@ -19,88 +18,228 @@ import {
+      },
+    });
+  };
+
+  saveWallet = async (newWallet: any) => {
     if (BlueApp.getWallets().some((wallet: Wallet) => wallet.getSecret() === newWallet.secret)) {
-      props.navigation.navigate(Route.ImportWallet);
-      setValidationError(i18n.wallets.importWallet.walletInUseValidationError);
+      this.setState({ validationError: i18n.wallets.importWallet.walletInUseValidationError });
     } else {
-      ReactNativeHapticFeedback.trigger('notificationSuccess', {
-        ignoreAndroidSystemSettings: false,
-      });
-      newWallet.setLabel(i18n.wallets.import.imported + ' ' + newWallet.typeReadable);
+      newWallet.setLabel(this.state.label || i18n.wallets.import.imported + ' ' + newWallet.typeReadable);
       BlueApp.wallets.push(newWallet);
       await BlueApp.saveToDisk();
-      props.loadWallets();
-      showSuccessImportMessageScreen();
+      this.props.loadWallets();
+      this.showSuccessImportMessageScreen();
+    }
+  };
+
+  createARWallet = (mnemonic: string) => {
+    try {
+      const wallet = new HDSegwitP2SHArWallet();
+      wallet.setMnemonic(mnemonic);
+      this.props.navigation.navigate(Route.IntegrateKey, {
+        onBarCodeScan: key => {
+          CreateMessage({
+            title: i18n.message.creatingWallet,
+            description: i18n.message.creatingWalletDescription,
+            type: MessageType.processingState,
+            asyncTask: () => this.saveARWallet(wallet, key),
+          });
+        },
+        title: i18n.wallets.publicKey.recoverySubtitle,
+        description: i18n.wallets.publicKey.recoveryDescription,
+        withLink: false,
+      });
+    } catch (_) {
+      this.showErrorMessageScreen();
     }
   };
 
-  const importMnemonic = async (mnemonic: string) => {
+  createAIRWallet = (mnemonic: string) => {
+    try {
+      const wallet = new HDSegwitP2SHAirWallet();
+      wallet.setMnemonic(mnemonic);
+      this.props.navigation.navigate(Route.IntegrateKey, {
+        onBarCodeScan: instantPublicKey => {
+          this.addRecoveryKey(wallet, instantPublicKey);
+        },
+        title: i18n.wallets.publicKey.recoverySubtitle,
+        description: i18n.wallets.publicKey.recoveryDescription,
+        withLink: false,
+      });
+    } catch (_) {
+      this.showErrorMessageScreen();
+    }
+  };
+
+  saveARWallet = async (wallet: HDSegwitP2SHArWallet, pubKeyHex: string) => {
+    try {
+      wallet.addPublicKey(pubKeyHex);
+      await wallet.generateAddresses();
+      await wallet.fetchBalance();
+      await wallet.fetchTransactions();
+      if (wallet.getBalance() > 0 || wallet.getTransactions().length !== 0) {
+        this.saveWallet(wallet);
+      } else {
+        this.showErrorMessageScreen();
+      }
+    } catch (error) {
+      this.showErrorMessageScreen();
+    }
+  };
+
+  saveAIRWallet = async (wallet: HDSegwitP2SHAirWallet, instantPublicKey: string, recoveryKey: string) => {
+    try {
+      wallet.addPublicKey(instantPublicKey);
```

Comment:
still, you don't validate every step of scanning, you have to validate recoveryPublicKey at first if it is not valid throw and error if valid navigate the user to scan instant public key